harness/determined/common/storage/boto3_credential_manager.py

```diff
@@ -0,0 +1,135 @@
+import logging
+import os
+import threading
+import time
+from typing import Any
+
+import boto3
+from botocore import session
+from botocore.compat import ensure_unicode
+from botocore.credentials import (
+    CredentialProvider,
+    Credentials,
+    ReadOnlyCredentials,
+    SharedCredentialProvider,
+)
+from botocore.session import get_session
+
+
+class RefreshedCredentialProvider(CredentialProvider):  # type: ignore
+    """
+    Creates a refreshable credential provider class given an existing credential provider in
+    the boto3 credential chain.
+
+    """
+
+    METHOD = "managed-refresh-cred"
+
+    def __init__(self, credential_provider: SharedCredentialProvider, check_every: int = 2) -> None:
+        super().__init__()
+        self.check_every = check_every
+        self.credential_provider = credential_provider
+
+    def load(self) -> Credentials:
+        return RefreshableSharedCredentials(
```

Comment:
I think you still need the `credentials and RefreshableSharedCredentials()` bit here, because returning None from this function is how the CredentialResolver knows to move on and call `load()` on the next CredentialProvider.